enterprise/cmd/frontend/internal/authz/resolvers/permission_sync_jobs.go

```
@@ -54,14 +44,43 @@ func (s *permissionSyncJobConnectionStore) ComputeNodes(ctx context.Context, arg
 
 	resolvers := make([]graphqlbackend.PermissionSyncJobResolver, 0, len(jobs))
 	for _, job := range jobs {
+		syncSubject, err := s.resolveSubject(ctx, job)
+		if err != nil {
+			return nil, err
+		}
 		resolvers = append(resolvers, &permissionSyncJobResolver{
-			db:  s.db,
-			job: job,
+			db:          s.db,
+			job:         job,
+			syncSubject: syncSubject,
 		})
 	}
 	return resolvers, nil
 }
 
+func (s *permissionSyncJobConnectionStore) resolveSubject(ctx context.Context, job *database.PermissionSyncJob) (graphqlbackend.PermissionSyncJobSubject, error) {
+	var repoResolver *graphqlbackend.RepositoryResolver
+	var userResolver *graphqlbackend.UserResolver
+
+	if job.UserID > 0 {
+		user, err := s.db.Users().GetByID(ctx, int32(job.UserID))
+		if err != nil {
+			return nil, err
+		}
+		userResolver = graphqlbackend.NewUserResolver(s.db, user)
+	} else {
```

What if both `UserID` and `RepositoryID` are `0`? Should we return an error in that case?

---

it is mutually exclusive via DB constraint.
Should not happen (c)